libraries/adsk/adsklib/adsklib_defs.mtlx

```diff
@@ -190,4 +190,16 @@
     <output name="out" type="color3" />
   </nodedef>
 
+  <!--
+    Node: token_node
+    Non-rendering node used to hold a set of token values
+  -->
+  <nodedef name="ND_token_node" node="token_node" nodegroup="organization" namespace="adsk">
```

Comment:
adsk only for now.

Reply:
I'm not sure why this node is needed? It seems it's a node just to plug tokens _into_, and there are not outputs and it's never used anywhere. Maybe you could give some more context why it's needed?

Reply:
Yes. This is the proposal to have a reusable node type which can hold a set of tokens.
It's more of a user workflow way to take a set of tokens and connect them up to as interfaces on nodegraphs and have it serializable.

There is no data flow per-se.

Reply:
Do you mean it's a way for the user to create tokens on a nodegraph interface, by creating a `token_node` and connecting its inputs/tokens to the nodegraphs interface to have them created there?

Because once created the tokens are part of the nodegraph interface, and their value is saved there. So after creation the `token_node` is no longer needed, right?

If this is the case it seems like it's a workaround for a limitation in the UI. Would it not be better so solve this in the UI instead and add support for creating/removing/editing tokens directly on a nodegraph's interface?

If it's a temporary workaround until VNN has this supported that's totally fine, but I'm not sure it's a good long term solution for creating tokens on an interface. Just let me know if I missed something and there is more to this than just a workaround for the UI.

Reply:
Hope this helps for clarity:

1. The base assumed functionality is tokens on a nodegraph which get applied withing the scope of the nodegraph.
As with inputs, tokens are made part of the interface and hence can have variants (presets) applied to them.
2. The intent of the token node is keep around reusable set of tokens to apply to only within nodegraph. i.e. they are locally scoped.
2a. The initial logic was to that just the existence of the token node within a nodegraph meant to apply all the tokens. 
2b. The current variant is to allow connections which means that to **only** use tokens from token node which have connections.  Basically allow for selective reuse of tokens from a token node.

Direct creation of tokens on a nodegraph isn't currently a UI requirement (and also not possible in VNN), though is possible without UI. (Fixes for that we're in the previous check-in).

Reply:
@bernardkwok Thanks for the additional details. I understand the intent better now.

The fact that existence of such a node overrides the tokens set on the nodegraph is a new concept I think. And I suppose this has to be cleared with Lucasfilm later as a proposal for the spec?

I find it a little bit peculiar to have a local override _inside_ a nodegraph that overrides something set outside (if a token is set on the nodegraph interface but also in a token_node internally). Just my 2 cents, and don't let this block the code review if you want to merge this in for now to work on it further.

Reply:
The proposal I put up a little while ago and awaiting feedback. 

I will try and provide more in the description and ping again. In the meantime, the unconnected case w/o a node I'll probably seperate out into a different check-in as those are fixes for already supported functionality.